drives.go

```diff
@@ -0,0 +1,65 @@
+package firecracker
+
+import (
+	"strconv"
+
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+)
+
+const rootDeviceName = "rootfs"
+
+// DrivesBuilder is a builder that will build an array of drives used to set up
+// the firecracker microVM. The DriveID will be an incrementing number starting
+// at one
+type DrivesBuilder struct {
+	rootDrive models.Drive
+	drives    []models.Drive
+}
+
+// NewDrivesBuilder will return a new DrivesBuilder with a given rootfs.
+func NewDrivesBuilder(rootfsPath string) DrivesBuilder {
+	return DrivesBuilder{}.WithRootFS(rootfsPath)
+}
+
+// DriveOpt represents an optional function used to allow for specific
+// customization of the models.Drive structure.
+type DriveOpt func(*models.Drive)
+
+// WithRootFS will set the given builder with the a new root path. The rootfs
+// will be set to read and write by default.
+func (b DrivesBuilder) WithRootFS(rootfsPath string, opts ...DriveOpt) DrivesBuilder {
```

Comment:
Is this strictly a (single) filesystem, or is this a device (or device image) that may contain a partition table with multiple filesystems inside?

Reply:
It is whatever firecracker's API supports. I know it supports a single filesystem, but where can I find a device image to test with?

Reply:
@ilianaw published a guide [here](https://iweller.people.a2z.com/con404/) ("Building an AMI").

Reply:
I think instead of testing, we could also just rename the function to be `WithRootDrive` instead of `WithRootFS` and be fine.  (Same thing for the `rootDeviceName` constant.)